Analysis/indiv_gals.py

```diff
@@ -1,95 +1,79 @@
-    bin_avg = bin_tbl[bin_type + '_avg'].data
-    bin_median = bin_tbl[bin_type + '_median'].data
     bin_idx = bin_npz['bin_ind']     #valid mass bin indices relative to unsorted data table 
     detect = valid_tbl[bin_names0[2]].data                   
     
     
-    #Get all bin valid values in 1D array
+    #Get all bin valid values in dictionary
+    bin_cols = [bin_names0[0], bin_names0[2], indv_names0[0]] + bin_mzevolve_names0
+    indiv_dict = {}
+    for name in bin_cols:
+        indiv_dict[name] = []
+    
     valid_idx = []
-    bin_ID_array = []
-    bin_min_array = []
-    bin_max_array = []
-    bin_avg_array = []
-    bin_median_array = []
-    detect_array = []
     for ii in range(len(bin_idx)):
         valid_idx += list(bin_idx[ii])
         for jj in range(len(bin_idx[ii])):
-            bin_ID_array.append(bin_ID[ii]) 
-            bin_min_array.append(bin_min[ii])
-            bin_max_array.append(bin_max[ii])
-            bin_avg_array.append(bin_avg[ii])
-            bin_median_array.append(bin_median[ii]) 
-            detect_array.append(detect[ii])            
-    idx_array = np.array(valid_idx)                     
+            indiv_dict[bin_names0[0]] += [bin_dict[bin_names0[0]][ii]]   #bin_ID
+            indiv_dict[bin_names0[2]] += [detect[ii]]                    #Detection
+            indiv_dict[bin_mzevolve_names0[0]] += [bin_dict[bin_mzevolve_names0[0]][ii]]   #logM_min
+            indiv_dict[bin_mzevolve_names0[1]] += [bin_dict[bin_mzevolve_names0[1]][ii]]   #logM_max
+            indiv_dict[bin_mzevolve_names0[2]] += [bin_dict[bin_mzevolve_names0[2]][ii]]   #logM_avg
+            indiv_dict[bin_mzevolve_names0[3]] += [bin_dict[bin_mzevolve_names0[3]][ii]]   #logM_median
+            indiv_dict[bin_mzevolve_names0[4]] += [bin_dict[bin_mzevolve_names0[4]][ii]]   #logLHb_min
+            indiv_dict[bin_mzevolve_names0[5]] += [bin_dict[bin_mzevolve_names0[5]][ii]]   #logLHb_max
+            indiv_dict[bin_mzevolve_names0[6]] += [bin_dict[bin_mzevolve_names0[6]][ii]]   #logLHb_avg
+            indiv_dict[bin_mzevolve_names0[7]] += [bin_dict[bin_mzevolve_names0[7]][ii]]   #logLHb_median
+    idx_array = np.array(valid_idx)
+    indiv_dict[indv_names0[0]] = objno[idx_array]      #ID     
 
     
-    out_ascii = fitspath + filename_dict['indv_bin_info']  
-    bin_cols = [bin_names0[0], bin_names0[2], indv_names0[0]]
-    bin_cols += [col for col in bin_mzevolve_names0 if col.startswith(bin_type)]
-    indiv_bin_info_tbl = Table([bin_ID_array, detect_array, objno[idx_array], bin_min_array, bin_max_array, bin_avg_array,
-                                bin_median_array], names = tuple(bin_cols))
-    asc.write(indiv_bin_info_tbl, out_ascii, format = 'fixed_width_two_line', overwrite = True)
+    out_ascii = fitspath + filename_dict['indv_bin_info']
+    asc.write(indiv_dict, names = tuple(bin_cols), output = out_ascii, format = 'fixed_width_two_line', overwrite = True)
```

Comment:
Might be able to drop the `names` input since you're passing in a dictionary.  We should check

Reply:
The Astropy documentation says that if you don't provide the "names" input, then the ordering of the columns would be unpredictable due to Python dictionaries not being ordered. So we could leave out "names," but the order of the columns in the table would change.